src/chai/ArrayManager.cpp

```diff
@@ -333,6 +333,11 @@ void ArrayManager::setUserCallback(void* pointer, UserCallback const& f)
   pointer_record->m_user_callback = f;
 }
 
+void ArrayManager::setUserCallback(UserCallback const& f)
```

Comment:
Everything looks good, my only suggestion would be to consider renaming this to `setGlobalUserCallback` to make it obvious what it's doing.

Reply:
Done.